connectivity/builder/egress_gateway_with_l7_policy.go

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package builder
+
+import (
+	_ "embed"
+
+	"github.com/cilium/cilium/pkg/versioncheck"
+
+	"github.com/cilium/cilium-cli/connectivity/check"
+	"github.com/cilium/cilium-cli/connectivity/tests"
+	"github.com/cilium/cilium-cli/utils/features"
+)
+
+//go:embed manifests/client-egress-icmp.yaml
+var clientEgressICMPYAML string
+
+//go:embed manifests/client-egress-l7-http-external-node.yaml
+var clientEgressL7HTTPExternalYAML string
+
+type egressGatewayWithL7Policy struct{}
+
+func (t egressGatewayWithL7Policy) build(ct *check.ConnectivityTest, _ map[string]string) {
+	newTest("egress-gateway-with-l7-policy", ct).
+		WithCondition(func() bool {
+			return versioncheck.MustCompile(">=1.16.0")(ct.CiliumVersion) && ct.Params().IncludeUnsafeTests
+		}).
+		WithCiliumPolicy(clientEgressICMPYAML).
```

Comment:
You must know this better than me, so I just ask for learning: why we need an ICMP policy for egress gateway?

Reply:
We check if traffic from pod to host is not masqueraded with an Egress IP by pinging host from the clients in the Egress Gateway test scenario.
https://github.com/cilium/cilium-cli/blob/1e249fd50f82d08ad0f60720417648b373420b02/connectivity/tests/egressgateway.go#L206-L219

In the egress-gateway-with-l7-policy test, we apply DNS policy and HTTP policy to all the client at egress and those endpoints go into the white list mode. So we need to explicitly allow the clients to send ICMP.